resource/logging_macros.h.em

```diff
@@ -219,152 +219,9 @@ typedef bool (* RclLogFilter)();
-    self.doc_lines = doc_lines
-
-feature_combinations = OrderedDict((
-  ('', Feature()),
-  ('_NAMED', Feature(
-    params=name_params,
-    args=name_args)),
-  ('_ONCE', Feature(
-    params=None,
-    args=once_args,
-    doc_lines=name_doc_lines)),
-  ('_ONCE_NAMED', Feature(
-    params=name_params,
-    args={**once_args, **name_args},
-    doc_lines=name_doc_lines)),
-  ('_EXPRESSION', Feature(
-    params=expression_params,
-    args=expression_args,
-    doc_lines=expression_doc_lines)),
-  ('_EXPRESSION_NAMED', Feature(
-    params=OrderedDict((*expression_params.items(), *name_params.items())),
-    args={**expression_args, **name_args},
-    doc_lines=expression_doc_lines + name_doc_lines)),
-  ('_FUNCTION', Feature(
-    params=function_params,
-    args=function_args,
-    doc_lines=function_doc_lines)),
-  ('_FUNCTION_NAMED', Feature(
-    params=OrderedDict((*function_params.items(), *name_params.items())),
-    args={**function_args, **name_args},
-    doc_lines=function_doc_lines + name_doc_lines)),
-  ('_SKIPFIRST', Feature(
-    params=None,
-    args=skipfirst_args,
-    doc_lines=skipfirst_doc_lines)),
-  ('_SKIPFIRST_NAMED', Feature(
-    params=name_params,
-    args={**skipfirst_args, **name_args},
-    doc_lines=skipfirst_doc_lines)),
-  ('_THROTTLE', Feature(
-    params=throttle_params,
-    args=throttle_args,
-    doc_lines=throttle_doc_lines)),
-  ('_SKIPFIRST_THROTTLE', Feature(
-    params=throttle_params,
-    args={
-      'condition_before': ' '.join([
-        throttle_args['condition_before'],
-        skipfirst_args['condition_before']]),
-      'condition_after': ' '.join([
-        throttle_args['condition_after'], skipfirst_args['condition_after']]),
-    },
-    doc_lines=skipfirst_doc_lines + throttle_doc_lines)),
-  ('_THROTTLE_NAMED', Feature(
-    params=OrderedDict((*throttle_params.items(), *name_params.items())),
-    args={**throttle_args, **name_args},
-    doc_lines=throttle_doc_lines)),
-  ('_SKIPFIRST_THROTTLE_NAMED', Feature(
-    params=OrderedDict((*throttle_params.items(), *name_params.items())),
-    args={
-      **{
-        'condition_before': ' '.join([
-          throttle_args['condition_before'],
-          skipfirst_args['condition_before']]),
-        'condition_after': ' '.join([
-          throttle_args['condition_after'],
-          skipfirst_args['condition_after']]),
-      }, **name_args
-    },
-    doc_lines=skipfirst_doc_lines + throttle_doc_lines)),
-))
-
-def get_macro_parameters(suffix):
-  return ''.join([p + ', ' for p in feature_combinations[suffix].params.keys()])
-
-def get_macro_arguments(suffix):
-  args = []
-  for k, default_value in default_args.items():
-    value = feature_combinations[suffix].args.get(k, default_value)
-    args.append(value)
-  return ''.join([a + ', ' for a in args])
+import sys
+sys.path.insert(0, rcutils_module_path)
+from rcutils.logging import feature_combinations, get_macro_arguments, get_macro_parameters, severities
```

Comment:
Use separate lines for each `import`.

Reply:
I squashed a change for this into https://github.com/ros2/rcutils/pull/49/commits/b71991f4a98340a4831fea52820da136ecfd4a37